src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -706,6 +706,15 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
         auto& weights = fc_node.weights();
         auto& input = fc_node.input();
         auto input_layout = input.get_output_layout();
+        // Change input data type of fully-connected node from i32 to f32
+        if (input_layout.data_type == data_types::i32) {
+            auto new_layout = input_layout;
+            new_layout.data_type = data_types::f32;
+            auto new_input = rf.get_reorder(input.id(), input_layout, new_layout);
+            if (new_input.first) {
+               p.add_intermediate(new_input.first, fc_node, 0);
+            }
+        }
```

Comment:
Alternatively, we could have added i32 to implementation_map. Does this change mean that the kernel operation does not support i32?

Reply:
Yes. Currently, all FC kernels do not support i32 input/output data type, so even if i32 is added to only to implementation map, the kernel selection does not work properly.

Reply:
OK I see. thanks for the explanation.